official/resnet/keras/keras_common.py

```diff
@@ -143,27 +167,53 @@ def get_optimizer():
 def get_callbacks(learning_rate_schedule_fn, num_images):
   """Returns common callbacks."""
   time_callback = keras_utils.TimeHistory(FLAGS.batch_size, FLAGS.log_steps)
-
-  tensorboard_callback = tf.keras.callbacks.TensorBoard(
-      log_dir=FLAGS.model_dir)
-
   lr_callback = LearningRateBatchScheduler(
       learning_rate_schedule_fn,
       batch_size=FLAGS.batch_size,
       num_images=num_images)
+  callbacks = [time_callback, lr_callback]
 
-  return time_callback, tensorboard_callback, lr_callback
+  if FLAGS.enable_tensorboard:
+    tensorboard_callback = tf.keras.callbacks.TensorBoard(
+        log_dir=FLAGS.model_dir)
+    callbacks.append(tensorboard_callback)
 
+  if FLAGS.profile_steps:
```

Comment:
Please make sure it does not include the step profiled by tensorboard callback.

Reply:
Good point. Since the TensorBoard profile step is customizable and it's not exposed by the class, I added a warning message when both callbacks are used. Observed that if they overlap, only the first one can start the profiler.

Reply:
SGTM